server_management/management/commands/backupdb.py

```diff
@@ -1,49 +1,47 @@
 from django.utils.timezone import now
-from fabric.api import local, settings, sudo
 
 from ._core import ServerManagementBaseCommand, load_config
 
 
-def perform_backup(env, config, remote):
-    with settings(warn_only=True):
-        # Dump the database on the server.
-        sudo('su - {user} -c \'pg_dump {name} -cOx -U {user} -f /home/{user}/{name}.sql --clean\''.format(
-            name=remote['database']['name'],
-            user=remote['database']['user'],
-        ))
-
-        backup_folder = '~/Backups/{}/{}'.format(
-            config['local']['database']['name'],
-            config['remote_name'],
-        )
-
-        # Create a backups folder
-        local('mkdir -p {}'.format(backup_folder))
-
-        # Pull the SQL file down.
-        local('scp {} {}@{}:/home/{}/{}.sql {}/{}.sql'.format(
-            '' if not getattr(env, 'key_filename') else ' -i {} '.format(env.key_filename),
-            env.user,
-            env.host_string,
-            remote['database']['user'],
-            remote['database']['name'],
-            backup_folder,
-            now().strftime('%Y%m%d%H%M'),
-        ))
-
-        # Delete the file on the server.
-        sudo('rm /home/{}/{}.sql'.format(
-            remote['database']['user'],
-            remote['database']['name'],
-        ))
+def perform_backup(config, connection):
+    remote = config['remotes'][config['remote_name']]
+
+    # Dump the database on the server.
+    connection.sudo('su - {user} -c \'pg_dump {name} -cOx -U {user} -f /home/{user}/{name}.sql --clean\''.format(
+        name=remote['database']['name'],
+        user=remote['database']['user'],
+    ))
+
+    backup_folder = '~/Backups/{}/{}'.format(
+        config['local']['database']['name'],
+        config['remote_name'],
+    )
+
+    # Create a backups folder
+    connection.local('mkdir -p {}'.format(backup_folder))
+
+    # Pull the SQL file down.
+    connection.local('scp {} {}@{}:/home/{}/{}.sql {}/{}.sql'.format(
+        f' -i {connection.connect_kwargs["key_filename"]} ' if connection.connect_kwargs.get('key_filename', False) else '',
```

Comment:
Assuming `connect_kwargs` is just a `dict`, I don't see the need for a `False` default here, since `.get` defaults to `None`.